CS/XSS: always escape output - 27

jrfnl opened the pull request:
## Summary

This PR can be summarized in the following changelog entry:
_Security hardening_

## Relevant technical choices:
* No functional changes.
* QA/CS compliance.

All output should be escaped, including translations.

Part of a series of PRs to fix these kind of issues.

Simple one(s). Use `esc_html()`, `esc_attr()` or `esc_url()` where appropriate.

PRs in this series include some function changes for `printf()` versus `echo sprintf()` and some function call layout changes.

** Remove superfluous variable setting, making it clear that the output is escaped when echo-ed out.


## Test instructions

Testing recommended, but no problems expected.

Test this by opening the relevant admin page in a browser and checking that the page layout has not been affected by this change.

andizer commented on admin/class-meta-columns.php:
```diff
@@ -102,13 +102,21 @@ public function column_content( $column_name, $post_id ) {
+				if ( '' === $metadesc_val ) {
+					echo '<span aria-hidden="true">&#8212;</span><span class="screen-reader-text">' . esc_html__( 'Meta description not set.', 'wordpress-seo' ) . '</span>';
+				}
+				else {
```
Is it possible to have an early return in the if? If so it might possible to remove the else

jrfnl replied:
Yes, that is absolutely possible.
I didn't do that as it makes it slightly less clear what's happening.

jrfnl replied:
@andizer Still want this ?

andizer replied:
Yes, please

jrfnl commented:
@andizer @moorscode Ok, rebased & fixed.

FYI: I've removed the `else`, but to maintain readability, I've reformatted the whole method.
* The method contained a *lot* of very long lines, I've broken those up now into logical units.
* The `switch` used `break`, but there was no other code after the `switch`, so changed these to `return` statements.
* Changed some unnecessary concatenation in `echo` statements to using comma's.

andizer commented:
CR + Acceptance done 👍
